nf_core/modules/module_test.py

```diff
@@ -56,6 +59,8 @@ def __init__(
         self.module_name = module_name
         self.no_prompts = no_prompts
         self.pytest_args = pytest_args
+        self.all_local_modules = ""
+        self.all_nfcore_modules = ""
```

Comment:
The use of empty strings as defaults here is a bit of a weird use case. Usually, these attributes contain lists, but empty lists should not be used as default, because they are mutable. Later in the code these attributes are looped over in list comprehensions. The empty string allows that and quietly produces an empty list when used in this fashion. While this works, it is a somewhat cryptic use of defaults. I like it better when a variable keeps its type. In this case I think it would be nicer to work with tuples than lists and empty tuples make predictable argument defaults. This is just a general comment, though and I would not block the PR because of it.

Reply:
Another way to be less cryptic is demonstrated in https://github.com/nf-core/tools/pull/1605/files